rpcs3/Emu/Cell/Modules/sceNp.cpp

```diff
@@ -3748,88 +4140,487 @@ error_code sceNpManagerSubSignout(vm::ptr<SceNpId> npId)
+    vm::ptr<SceNpMatchingAttr> attribute, vm::ptr<SceNpMatchingGUIHandler> handler, int param_7, int param_8)
+{
+	if (!communicationId || !range || !handler)
+		return SCE_NP_MATCHING_ERROR_INVALID_ARG;
+
+	if ((!param_8 && !range->start) || range->max > 20)
+		return SCE_NP_MATCHING_ERROR_INVALID_ARG;
+
+	u32 total_count = 0;
+	u32 inequality_count = 0;
+	u32 max_count = 10;
+	if (param_7 != 0)
+	{
+		max_count = 9;
+	}
+
+	for (auto con = cond; !!con; con = con->next)
+	{
+		if (++total_count > max_count)
+			return SCE_NP_MATCHING_ERROR_COND_MAX;
+
+		if (con->comp_type != SCE_NP_MATCHING_CONDITION_TYPE_VALUE ||
+			con->comp_op < SCE_NP_MATCHING_CONDITION_SEARCH_EQ ||
+			con->comp_op > SCE_NP_MATCHING_CONDITION_SEARCH_GE)
+			return SCE_NP_MATCHING_ERROR_INVALID_COND;
+
+		if (con->comp_op > SCE_NP_MATCHING_CONDITION_SEARCH_NE &&
+			++inequality_count > max_count)
+			return SCE_NP_MATCHING_ERROR_COMP_OP_INEQUALITY_MAX;
+
+		switch (con->target_attr_type)
+		{
+		case SCE_NP_MATCHING_ATTR_TYPE_BASIC_BIN:
+		case SCE_NP_MATCHING_ATTR_TYPE_GAME_BIN:
+		{
+			return SCE_NP_MATCHING_ERROR_INVALID_COND;
+		}
+		case SCE_NP_MATCHING_ATTR_TYPE_BASIC_NUM:
+		{
+			if (param_7 == 0)
+			{
+				if (con->target_attr_id < SCE_NP_MATCHING_ROOM_ATTR_ID_TOTAL_SLOT ||
+					con->target_attr_id > SCE_NP_MATCHING_ROOM_ATTR_ID_ROOM_SEARCH_FLAG)
+					return SCE_NP_MATCHING_ERROR_INVALID_COND;
+			}
+			else
+			{
+				if (con->target_attr_id == SCE_NP_MATCHING_ROOM_ATTR_ID_TOTAL_SLOT)
+					return SCE_NP_MATCHING_ERROR_INVALID_COND;
+			}
+			break;
+		}
+		case SCE_NP_MATCHING_ATTR_TYPE_GAME_NUM:
+		{
+			const u32 max_id = (param_7 == 0) ? 16 : 8;
+			if (con->target_attr_id < 1 || con->target_attr_id > max_id)
+				return SCE_NP_MATCHING_ERROR_INVALID_COND;
+			break;
+		}
+		default:
+			break;
+		}
+	}
+
+	return check_attr_id_and_duplicate(attribute);
+}
+
 error_code sceNpMatchingGetRoomListLimitGUI(u32 ctx_id, vm::ptr<SceNpCommunicationId> communicationId, vm::ptr<SceNpMatchingReqRange> range, vm::ptr<SceNpMatchingSearchCondition> cond,
     vm::ptr<SceNpMatchingAttr> attr, vm::ptr<SceNpMatchingGUIHandler> handler, vm::ptr<void> arg)
 {
-	sceNp.todo(
-	    "sceNpMatchingGetRoomListLimitGUI(ctx_id=%d, communicationId=*0x%x, range=*0x%x, cond=*0x%x, attr=*0x%x, handler=*0x%x, arg=*0x%x)", ctx_id, communicationId, range, cond, attr, handler, arg);
+	sceNp.todo("sceNpMatchingGetRoomListLimitGUI(ctx_id=%d, communicationId=*0x%x, range=*0x%x, cond=*0x%x, attr=*0x%x, handler=*0x%x, arg=*0x%x)", ctx_id, communicationId, range, cond, attr, handler, arg);
+
+	if (false) // TODO
+		return SCE_NP_MATCHING_ERROR_UTILITY_UNAVAILABLE;
+
+	// This check is used in all sceNpMatchingGetRoomList functions
+	error_code err = check_room_list_params(ctx_id, communicationId, range, cond, attr, handler, 1, 0);
```

Comment:
Any purpose to keeping the last 2 params if they're fixed?

Reply:
check_room_list_params is also used in sceNpMatchingGetRoomList with at least one different parameter